### ISSUE TITLE
@auth/core Attempted import error: 'experimental_customFetch' is not exported from 'oauth4webapi'

### ISSUE THREAD

**Issue by woofers** [nextauthjs/next-auth#10979]
### Environment

```
  System:
    OS: Linux 6.5 Ubuntu 20.04.6 LTS (Focal Fossa)
    CPU: (2) x64 Intel(R) Xeon(R) Platinum 8370C CPU @ 2.80GHz
    Memory: 6.01 GB / 7.74 GB
    Container: Yes
    Shell: 5.0.17 - /bin/bash
  Binaries:
    Node: 20.12.1 - ~/nvm/current/bin/node
    Yarn: 1.22.19 - /usr/bin/yarn
    npm: 10.5.0 - ~/nvm/current/bin/npm
    pnpm: 8.15.6 - ~/nvm/current/bin/pnpm
  npmPackages:
    @auth/unstorage-adapter: ^2.0.0 => 2.1.0 
    next: latest => 14.2.3 
    next-auth: beta => 5.0.0-beta.18 
    react: ^18.2.0 => 18.3.1 
 ```

### Reproduction URL

https://github.com/woofers/next-auth-oauth4webapi

### Describe the issue

When `oauth4webapi` is installed with any version between `v2.4.0` up to `v2.6.0`, no export exists for `experimental_customFetch`, as this was added in `v2.6.0`.  This leads to `next build` failing when certain versions of `oauth4webapi` allowed by semver are installed.

### How to reproduce

1. Go to example repo: https://github.com/woofers/next-auth-oauth4webapi
2. Run `pnpm install` and `pnpm build`
3. Notice build fails with:

```
./node_modules/.pnpm/@auth+core@0.31.0/node_modules/@auth/core/lib/actions/callback/oauth/callback.js
Attempted import error: 'experimental_customFetch' is not exported from 'oauth4webapi' (imported as 'o').

Import trace for requested module:
./node_modules/.pnpm/@auth+core@0.31.0/node_modules/@auth/core/lib/actions/callback/oauth/callback.js
./node_modules/.pnpm/@auth+core@0.31.0/node_modules/@auth/core/lib/actions/callback/index.js
./node_modules/.pnpm/@auth+core@0.31.0/node_modules/@auth/core/lib/actions/index.js
./node_modules/.pnpm/@auth+core@0.31.0/node_modules/@auth/core/lib/index.js
./node_modules/.pnpm/@auth+core@0.31.0/node_modules/@auth/core/index.js
./node_modules/.pnpm/next-auth@5.0.0-beta.18_next@14.2.3_react@18.3.1/node_modules/next-auth/index.js
```

-----

To repro all that needs to happen is for pnpm to resolve a compatible version of `oauth4webapi` that is allowed by the `@auth/core` semver range:
- Either happens automatically due to pnpm resolution
- Can be forced by manually adding to project `package.json`: https://github.com/woofers/next-auth-oauth4webapi/commit/812d153a75414c9cf4ad23f4682e8e0b15f2641f - In the case `oauth4webapi` is required by another library or your own project

### Expected behavior

`next build` should pass and `oauth4webapi` should work for all semver ranges.

**Comment by woofers** [comment]
I have a PR fix open for this issue: https://github.com/nextauthjs/next-auth/pull/10978

**Comment by woofers** [comment]
This is now resolved with #10978
